src/Documentation/RstUtils.php

```diff
@@ -33,6 +33,15 @@ public static function toRst(string $string, int $indent = 0): string
         return 0 === $indent ? $string : self::indent($string, $indent);
     }
 
+    public static function toInlineCode(string $string, bool $shouldAddBackticks = true): string
+    {
+        if (str_contains($string, '`')) {
+            return str_replace('`', '``', $string);
+        }
+
+        return $shouldAddBackticks ? '``'.$string.'``' : $string;
```

Comment:
I believe this implementation is too optimistic. If there's already string with double backtick, it would be changed to four. Not to mention `$shouldAddBackticks` is not used at all, because both usages of `toInlineCode()` don't pass second argument, so default is used, ergo: it could be omitted and simplified to:

```php
return str_contains($string, '`') ? return str_replace('`', '``', $string) : '``'.$string.'``';
```

But it's only a side note, since I think it would be better to just extract `Preg::replace()` from `toRst()` into `public function ensureProperInlineCode(string $string)`, and then use this method for rendering descriptions in `RuleSetDocumentationGenerator::generateRuleSetsDocumentation()`. Then rendering fixer's name could be untouched.

You can see [here](https://regex101.com/r/lqdOeG/3) that for single backticks it will wrap content with double backticks, and for double backticks it would also do 4, so we could change regex to:

```
(?<!`)(`[^`]+`)(?!`)
```

and ensure only single backticks are fixed ([see in action](https://regex101.com/r/lqdOeG/2)).

Reply:
@Wirone 
Your suggestion is great, there are only two situations that need to be corrected: 
1. Where the X method is currently used; 
2. In all description.
My previous practice is wrong.